packages/discovery-react-components/src/components/StructuredQuery/components/AddRuleGroupButton/AddRuleGroupButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FC, Dispatch, SetStateAction } from 'react';
+import keys from 'lodash/keys';
+import { Button } from 'carbon-components-react';
+import Add16 from '@carbon/icons-react/lib/add/16';
+import { Messages } from 'components/StructuredQuery/messages';
+import { StructuredQuerySelection } from 'components/StructuredQuery/utils/structuredQueryInterfaces';
+
+export interface AddRuleGroupButtonProps {
+  /**
+   * text to display for the Add rule group button
+   */
+  addRuleGroupText: Messages['addRuleGroupText'];
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  structuredQuerySelection: StructuredQuerySelection;
+  /**
+   * used to set the structuredQuerySelection state
+   */
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>;
+}
+
+export const AddRuleGroupButton: FC<AddRuleGroupButtonProps> = ({
+  addRuleGroupText,
+  structuredQuerySelection,
+  setStructuredQuerySelection
+}) => {
+  const handleOnClick = () => {
+    const maxGroupId = keys(structuredQuerySelection.groups).reduce(function(a, b) {
+      return Math.max(parseInt(a), parseInt(b)).toString();
```

Comment:
does this need to be `toString()`-ed?

Reply:
Yes, otherwise TS complains because `string` is the return type expected here...